### PR TITLE
James server test junit5 (part 3)

### PR DESCRIPTION
**Pull request by chibenwa**
See https://github.com/linagora/james-project/pull/1766 & https://github.com/linagora/james-project/pull/1763 first

Also a rebase would be needed on top of https://github.com/linagora/james-project/pull/1763

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[18b3fb072279c1e4fddbc1afb2f5ae10b3036414] [ERROR]   ESReporterTest.timeMetricsShouldBeReportedWhenImapCommandsReceived:121 » ConditionTimeout
[18b3fb072279c1e4fddbc1afb2f5ae10b3036414] [ERROR]   ESReporterTest.timeMetricsShouldBeReportedWhenJmapRequestsReceived:143 » ConditionTimeout
```

**Comment by chibenwa** [comment]
Force pushed to solve conflict

**Comment by chibenwa** [comment]
```
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.573 s - in org.apache.james.queue.rabbitmq.view.cassandra.model.BucketedSlicesTest
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [INFO] 
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [INFO] Results:
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [INFO] 
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [ERROR] Failures: 
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] [ERROR]   RabbitMQMailQueueTest.browseShouldReturnCurrentlyEnqueuedMailFromAllSlices:176 
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] Expecting:
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]   <["1-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-5"]>
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] to contain exactly (and in same order):
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]   <["1-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "1-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "2-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "3-5",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-1",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-2",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-3",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-4",
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]     "5-5"]>
[9dde8a51ae7a86584b109ddc51d432c4e1af4268] but could not find the following elements:
[9dde8a51ae7a86584b109ddc51d432c4e1af4268]   <["5-4"]>
```

test this please

**Comment by chibenwa** [comment]
I do not succeed to reproduce last failure locally...

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/modules/CassandraJMAPTestModule.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.webadmin.WebAdminConfiguration;
+
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+
+public interface CassandraJMAPTestModule {
+    Module DEFAULT = Modules.override(ALL_BUT_JMX_CASSANDRA_MODULE)
+        .with(TestJMAPServerModule.DEFAULT,
+            binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class));
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/modules/CassandraJMAPTestModule.java]: why do we need this specific extract in the first place ?

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/modules/CassandraJMAPTestModule.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    Module ENABLE_WEBADMIN = binder -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION);
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/modules/CassandraJMAPTestModule.java]: UNPROTECTED_WEBADMIN ?

File: server/container/guice/protocols/jmap/src/test/java/org/apache/james/modules/TestJMAPServerModule.java [hunk 3]
```diff
@@ -33,6 +33,8 @@
 import com.google.inject.name.Names;
 
 public class TestJMAPServerModule extends AbstractModule {
+    public static final int MAXIMUM_MESSAGE_COUNT = 10;
+    public static final TestJMAPServerModule DEFAULT = new TestJMAPServerModule(MAXIMUM_MESSAGE_COUNT);
```
Comment (mbaechler) [inline server/container/guice/protocols/jmap/src/test/java/org/apache/james/modules/TestJMAPServerModule.java]: why do we need this 10 messages limitation everywhere ?

Comment (chibenwa) [reply server/container/guice/protocols/jmap/src/test/java/org/apache/james/modules/TestJMAPServerModule.java]: It is used to provide a default value when user do not provide limits for GET MESSAGE LIST.

In prod a constant bound to 256 is used.

So no, we do not even need it...

**Comment by openpaas-james** [comment]
Build finished. No test results found.

**Comment by chibenwa** [comment]
These tests now rely on JUNIT 5

**Pull request closed without merging**